googlepay/src/main/java/com/airwallex/android/googlepay/GooglePayComponent.kt

```diff
@@ -23,11 +46,75 @@ class GooglePayComponent : ActionComponent {
         cardNextActionModel: CardNextActionModel?,
         listener: Airwallex.PaymentResultListener
     ) {
-        TODO("Not yet implemented")
+        this.paymentIntentId = paymentIntentId
+        this.listener = listener
+        val googlePayOptions = session.googlePayOptions ?: return
+        val paymentDataRequestJson = PaymentsUtil.getPaymentDataRequest(
+            price = session.amount,
+            countryCode = session.countryCode,
+            currency = session.currency,
+            googlePayOptions = googlePayOptions,
+            supportedCardSchemes = paymentMethodType.cardSchemes
+        ) ?: run {
+            listener.onCompleted(
+                AirwallexPaymentStatus.Failure(
+                    AirwallexCheckoutException(
+                        message = "Can't serialize Google Pay payment data request"
+                    )
+                )
+            )
+            return
+        }
+        val request = PaymentDataRequest.fromJson(paymentDataRequestJson.toString())
+        val paymentClient = PaymentsUtil.createPaymentsClient(activity)
+        resolvePaymentRequest(
+            paymentClient.loadPaymentData(request),
+            activity,
+            loadPaymentDataRequestCode
+        )
     }
 
     override fun handleActivityResult(requestCode: Int, resultCode: Int, data: Intent?): Boolean {
-        TODO("Not yet implemented")
+        if (requestCode == loadPaymentDataRequestCode) {
+            when (resultCode) {
+                RESULT_OK -> {
+                    val id = paymentIntentId ?: run {
+                        Logger.error(
+                            errorTag,
+                            "Invalid payment intent ID"
+                        )
+                        listener?.onCompleted(AirwallexPaymentStatus.Cancel)
+                        return true
+                    }
+                    data?.let { intent ->
+                        PaymentData.getFromIntent(intent)?.let(::createPaymentSuccessInfo)?.let {
+                            listener?.onCompleted(AirwallexPaymentStatus.Success(id, it))
```

Comment:
Yeah these `lets` are very hard to read :P. Can we improve this?

Reply:
yep, just refactored using `if else`